test/integration/apiserver/apply/benchmark_test.go

```diff
@@ -0,0 +1,385 @@
+    type: Initialized
+  - lastProbeTime: null
+    lastTransitionTime: "2019-07-08T09:41:59Z"
+    status: "True"
+    type: Ready
+  - lastProbeTime: null
+    lastTransitionTime: null
+    status: "True"
+    type: ContainersReady
+  - lastProbeTime: null
+    lastTransitionTime: "2019-07-08T09:31:18Z"
+    status: "True"
+    type: PodScheduled
+  containerStatuses:
+  - containerID: docker://885e82a1ed0b7356541bb410a0126921ac42439607c09875cd8097dd5d7b5376
+    image: some-image-name
+    imageID: docker-pullable://some-image-id
+    lastState:
+      terminated:
+        containerID: docker://d57290f9e00fad626b20d2dd87a3cf69bbc22edae07985374f86a8b2b4e39565
+        exitCode: 255
+        finishedAt: "2019-07-08T09:39:09Z"
+        reason: Error
+        startedAt: "2019-07-08T09:38:54Z"
+    name: name
+    ready: true
+    restartCount: 6
+    state:
+      running:
+        startedAt: "2019-07-08T09:41:59Z"
+  hostIP: 10.0.0.1
+  phase: Running
+  podIP: 10.0.0.1
+  qosClass: BestEffort
+  startTime: "2019-07-08T09:31:18Z"
+`)
+
+func decodePod(podBytes []byte) v1.Pod {
+	pod := v1.Pod{}
+	err := yaml.Unmarshal(podBytes, &pod)
+	if err != nil {
+		panic(err)
+	}
+	return pod
+}
+
+func encodePod(pod v1.Pod) []byte {
+	podBytes, err := yaml.Marshal(pod)
+	if err != nil {
+		panic(err)
+	}
+	return podBytes
+}
+
+func BenchmarkNoServerSideApply(b *testing.B) {
+	defer featuregatetesting.SetFeatureGateDuringTest(b, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, false)()
+
+	_, client, closeFn := setup(b)
+	defer closeFn()
+	flag.Lookup("v").Value.Set("0")
+
+	benchAll(b, client, decodePod(podBytes))
+}
+
+func getPodSizeWhenEnabled(b *testing.B, pod v1.Pod) int {
+	return len(getPodBytesWhenEnabled(b, pod, "application/vnd.kubernetes.protobuf"))
+}
+
+func getPodBytesWhenEnabled(b *testing.B, pod v1.Pod, format string) []byte {
+	defer featuregatetesting.SetFeatureGateDuringTest(b, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+	_, client, closeFn := setup(b)
+	defer closeFn()
+	flag.Lookup("v").Value.Set("0")
+
+	pod.Name = "size-pod"
+	podB, err := client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		Name(pod.Name).
+		Namespace("default").
+		Param("fieldManager", "apply_test").
+		Resource("pods").
+		SetHeader("Accept", format).
+		Body(encodePod(pod)).DoRaw(context.TODO())
+	if err != nil {
+		b.Fatalf("Failed to create object: %#v", err)
+	}
+	return podB
+}
+
+func BenchmarkNoServerSideApplyButSameSize(b *testing.B) {
+	pod := decodePod(podBytes)
+
+	ssaPodSize := getPodSizeWhenEnabled(b, pod)
+
+	defer featuregatetesting.SetFeatureGateDuringTest(b, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, false)()
+	_, client, closeFn := setup(b)
+	defer closeFn()
+	flag.Lookup("v").Value.Set("0")
+
+	pod.Name = "size-pod"
+	noSSAPod, err := client.CoreV1().RESTClient().Post().
+		Namespace("default").
+		Resource("pods").
+		SetHeader("Content-Type", "application/yaml").
```

Comment:
potentially the media type could have been take from https://github.com/kubernetes/kubernetes/blob/bce9d225b1958e1c393bfcee149706cbdf3a3581/staging/src/k8s.io/apimachinery/pkg/runtime/types.go

but we have this in other tests already, so it seems fine to use the literal too.

Reply:
This is not new, just moving the test from one file to another :-/